go-controller/pkg/crd/routeadvertisements/v1/types.go

```diff
@@ -0,0 +1,91 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:path=routeadvertisements,scope=Cluster,shortName=ra,singular=routeadvertisements
+// +kubebuilder::singular=routeadvertisements
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=".status.status"
+// RouteAdvertisements is the Schema for the routeadvertisements API
+type RouteAdvertisements struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RouteAdvertisementsSpec   `json:"spec,omitempty"`
+	Status RouteAdvertisementsStatus `json:"status,omitempty"`
+}
+
+// RouteAdvertisementsSpec defines the desired state of RouteAdvertisements
+// +kubebuilder:validation:XValidation:rule="!has(self.nodeSelector) || !('PodNetwork' in self.advertisements)",message="If 'PodNetwork' is selected for advertisement, a 'nodeSelector' can't be specified as it needs to be advertised on all nodes"
+type RouteAdvertisementsSpec struct {
```

Comment:
So this just occurred to me...sorry this is so late. We have this RouteAdvertisementsSpec, which I think is good as it is generic and indicates we want to advertise or import routes with a dynamic routing protocol. We are also tied to FRR here, which is also OK imo. However, with this configuration, we are assuming BGP. I wonder if it would be good to add a Protocol field and set it to a type "BGP". Maybe since we only support BGP for now it could be added later, but I just wanted to bring it up now.

Reply:
We can more fields later. But wouldn't that be something that could potentially be already indicated or inferred from the selected FRRConfigurations? Difficult to say without looking how a different routing protocol would be implemented in FRR-K8s.

Reply:
yeah, I was only thinking from the API perspective. We can introduce a new field "Type" or "Protocol" later, but we will have to default to BGP since that was what we assume in the CRD today. I think that is fine though.